src/event/cloudevent.ts

```diff
@@ -167,6 +168,17 @@ export class CloudEvent implements CloudEventV1, CloudEventV03 {
     }
   }
 
+  /**
+   * Emit this CloudEvent through the application
+   *
+   * @param {boolean} ensureDelivery fail the promise if one listener fail
+   * @return {Promise<CloudEvent>} this
+   */
+  public async emit(ensureDelivery = false): Promise<this> {
```

Comment:
Not a fan of `ensureDelivery`. I would expect no parameter. The Promise should fail if there is no delivery and the user should retry the delivery or handle it.

Reply:
The event system of node is not done this way, so if we follow normal events it will never fail, I think adding the option to ensure delivery is useful for some cases